File: seed/csharp-sdk/trace/src/SeedTrace/V2/Problem/Types/CreateProblemRequestV2.cs
using System.Text.Json.Serialization;
using SeedTrace;
using SeedTrace.V2;

#nullable enable

namespace SeedTrace.V2;

public record CreateProblemRequestV2
{
    [JsonPropertyName("problemName")]
    public required string ProblemName { get; }

    [JsonPropertyName("problemDescription")]
    public required ProblemDescription ProblemDescription { get; }

    [JsonPropertyName("customFiles")]
    public required object CustomFiles { get; }

    [JsonPropertyName("customTestCaseTemplates")]
    public IEnumerable<TestCaseTemplate> CustomTestCaseTemplates { get; } =
        new List<TestCaseTemplate>();

    [JsonPropertyName("testcases")]
    public IEnumerable<TestCaseV2> Testcases { get; } = new List<TestCaseV2>();

    [JsonPropertyName("supportedLanguages")]
    public HashSet<Language> SupportedLanguages { get; } = new HashSet<Language>();

    [JsonPropertyName("isPublic")]
    public required bool IsPublic { get; }
}

File: seed/ruby-sdk-v2/trace/lib/create_problem_request_v_2.rb
# frozen_string_literal: true

module V2
    module Types
        class CreateProblemRequestV2 < Internal::Types::Model
            field :problem_name, String, optional: true, nullable: true
            field :problem_description, ProblemDescription, optional: true, nullable: true
            field :custom_files, CustomFiles, optional: true, nullable: true
            field :custom_test_case_templates, Array, optional: true, nullable: true
            field :testcases, Array, optional: true, nullable: true
            field :supported_languages, Array, optional: true, nullable: true
            field :is_public, Boolean, optional: true, nullable: true
        end
    end
end
